internal/gengapic/auxiliary.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	longrunning "cloud.google.com/go/longrunning/autogen/longrunningpb"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// auxTypes gathers details of types we need to generate along with the client
+type auxTypes struct {
+	// Map of RPC descriptor to assigned operationWrapper.
+	methodToWrapper map[*descriptorpb.MethodDescriptorProto]operationWrapper
```

Comment:
Someday we should create a lightweight generic sorted map impl. We always be sorting in generator because of key randomness today.

Reply:
Oo that would be great, and fun :)

If you want to, we could start one in this repo's internal package, then potentially give it its own repo? Would be good practice with generics...